lightning/src/ln/outbound_payment.rs

```diff
@@ -1045,6 +1045,11 @@ impl OutboundPayments {
 						abandon_with_entry!(entry, PaymentFailureReason::UnknownRequiredFeatures);
 						return Err(Bolt12PaymentError::UnknownRequiredFeatures)
 					}
+					if duration_since_epoch > invoice.created_at().saturating_add(invoice.relative_expiry()) {
```

Comment:
Note for reviewers: for inbound payments, we buffer expiries by 2 hours to account for inaccurate block times in `no-std`. So we could do that here too, but it seemed maybe unnecessary since static invoices are expected to have quite long expiries... Let me know if we'd prefer to do that. (Our other outbound expiry checks are `std`-only, but I wanted `no-std` here for ease of testing).

Reply:
I looked for a bit and couldn't find a good place to put the util (`invoice_utils` might make sense but it's being deprecated IIUC?). So I might just leave this as-is until/unless we refactor outbound_payments to detect expired invoices in `no-std` more generally.